rust/README.md

```diff
@@ -3,16 +3,14 @@
 A training course on practical software construction in Rust.
 
 The final project in this course is a networked, persistent [key-value store]
-with multithreading, asynchronous I/O (via [tokio] and [tower]), a
-high-performance network protocol ([GRPC] via [prost] and [tower-grpc]),
-backed by a simple [LSM-tree] that the student will write themselves.
+with multithreading, asynchronous I/O (via [hyper]).
 
 Subjects covered include:
 
 - Structuring and maintaining Rust programs
 - Fun and foolproof parallel programming
-- Asyncronous programming with [futures] and [tokio]
-- High-performance networking with [GRPC] and [prost]
+- Asyncronous programming with [futures]
+- Networking with [hyper]
```

Comment:
I wasn't planning to do anything HTTP2 specific, though I assume hyper will negotiate a HTTP2 upgrade if it can. Do you have any suggestions?

I'm torn about using HTTP for this, since gRPC is what PingCAP uses. But @siddontang and I discussed that it may be too much to require gRPC for this project. If we don't do gRPC in the initial set of projects, I'm thinking it could be in a subsequent project.

Reply:
@nrc Somewhere @sticnarf suggested having the student just implement basic HTTP for that project, which does force them to write "real" networking code (vs just dropping in hyper).